OA Report input and aggregate label view

Irisvella opened the pull request:
- #347 OA Report input to create orders, orderParts as xlsm or csv 
To access this, go on nav bar and click -> OA Report 
![image](https://github.com/user-attachments/assets/e373ada9-2cc7-43c4-ada1-8628661035b2)

- View all labels and corrected labels according to stakeholder specification. 
To access this: view inventory pick lists -> view labels 
![image](https://github.com/user-attachments/assets/8f46f2c0-c379-45ee-a3e5-01a9cc22c01c)

- Updated inventory picklist items UI and backend DB fetching to display and demo 
![image](https://github.com/user-attachments/assets/080f3e37-6788-441b-9874-898084fe1cf1)

Irisvella committed:
#347 FE fix and added navbar navigation

- FE fix and added navbar navigation

Irisvella committed:
#347 Create orders and order parts from uploaded OA report

- From OA Report, orders and orderparts are automatically created.
- Can check for duplicates later

Irisvella committed:
Merge branch 'main' into andy-iteration-12-fixes

Irisvella committed:
Create 0010_merge_20250323_2325.py

Irisvella committed:
#347 Updated InventoryPicklist and PicklistItem generation logic

- Updated inventory picklist and picklistitem creation logic to include new information. Location pending stakeholder info.
- Adjusted labels to correctly fetch info

Irisvella committed:
update label font

Irisvella committed:
enforce unique constraint for orderparts upload

enforce unique constraint for orderparts upload

Irisvella committed:
display all labels + print

- All labels are displayed for an order and print function enabled

Irisvella committed:
Add accessibility to viewing all labels in UI

Add accessibility to viewing all labels in UI

Irisvella committed:
Making columns more visually distinctive

As per stakeholder request #347

Irisvella committed:
corrected small column name mixup

Irisvella committed:
added tooltip for OAReport input

#347 Tooltip

Irisvella committed:
Revert "Merge pull request #200 from Irisvella/task-107-limit-page-view-by-role"

This reverts commit b19fd4db42484332b3417eb501dac453a7f9a733, reversing
changes made to 9a97d92a716b1453ddc8c5c6cf8d72df96a1271f.

Irisvella committed:
fix tests

Irisvella committed:
re-add view labels button

Irisvella committed:
corrected typo in labelmaker

Irisvella committed:
Update tests.py

Irisvella commented:
If the backend breaks, you likely will need to makemigrations and migrate since there were some DB changes using another .env

NourHadieHassoun reviewed (approved):
Very useful!!

m-coscia reviewed (changes requested):
@Irisvella The picklist labels dont show up
![image](https://github.com/user-attachments/assets/c40becf1-d4a1-4829-b98a-6e91e8cab21f)
and I have tried to see if there are any db changes to migrate and that didnt work. does this only work on the new db?

Irisvella commented:
> @Irisvella The picklist labels dont show up ![image](https://private-user-images.githubusercontent.com/95299392/428410741-c40becf1-d4a1-4829-b98a-6e91e8cab21f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0NDcwMDMsIm5iZiI6MTc0MzQ0NjcwMywicGF0aCI6Ii85NTI5OTM5Mi80Mjg0MTA3NDEtYzQwYmVjZjEtZDRhMS00ODI5LWI5OGEtNmU5MWU4Y2FiMjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzMxVDE4NDUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4NWM3MTM4ZWU3NzM3ZTE1YzgyNzA0YzU4NzBiZTI3MTE1MTUyOWNhODM5NjQ1NjEzZmFkMzZiYTkyZTUwODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sCOvMXl7MOwjHw4uhCRbXzyur7_E-y-jvWG9mBxu0VM) and I have tried to see if there are any db changes to migrate and that didnt work. does this only work on the new db?

yeah it's better to use a fresh DB with up to date info

Irisvella committed:
#347 additional data preprocessing

- Additional data preprocessing to remove N/A and rows with bad data.
- Still better to upload in small chunks. It seems to fail when using all 16k lines

m-coscia commented:
> > @Irisvella The picklist labels dont show up ![image](https://private-user-images.githubusercontent.com/95299392/428410741-c40becf1-d4a1-4829-b98a-6e91e8cab21f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM0NDcwMDMsIm5iZiI6MTc0MzQ0NjcwMywicGF0aCI6Ii85NTI5OTM5Mi80Mjg0MTA3NDEtYzQwYmVjZjEtZDRhMS00ODI5LWI5OGEtNmU5MWU4Y2FiMjFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMzElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzMxVDE4NDUwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU4NWM3MTM4ZWU3NzM3ZTE1YzgyNzA0YzU4NzBiZTI3MTE1MTUyOWNhODM5NjQ1NjEzZmFkMzZiYTkyZTUwODgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sCOvMXl7MOwjHw4uhCRbXzyur7_E-y-jvWG9mBxu0VM) and I have tried to see if there are any db changes to migrate and that didnt work. does this only work on the new db?
> 
> yeah it's better to use a fresh DB with up to date info

alright, i'll remove the request

m-coscia reviewed (approved):
good job on creating labels!
